File: src/0/dp/1014/S1014_v1.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        imax = values[0]
        res = 0
        for i in values[1:]:
            imax -= 1
            if (s := imax + i) > res:
                res = s
            if i > imax:
                imax = i
        return res

File: src/0/dp/1014/S1014_v1.java
class Solution {
    int imax, tmp, res;
    public int maxScoreSightseeingPair(int[] values) {
        imax = 0;
        res = 0;
        for (int i : values) {
            tmp = imax + i;
            if (tmp > res) res = tmp;
            if (i > imax) imax = i;
            imax--;
        }
        return res;
    }
}
